shotover-proxy/src/transforms/cassandra/sink_cluster/node_pool.rs

```diff
@@ -0,0 +1,88 @@
+use super::routing_key::calculate;
+use super::token_map::TokenMap;
+use crate::transforms::cassandra::sink_cluster::node::CassandraNode;
+use cassandra_protocol::frame::message_execute::BodyReqExecuteOwned;
+use cassandra_protocol::frame::message_result::PreparedMetadata;
+use cassandra_protocol::frame::Version;
+use cassandra_protocol::token::Murmur3Token;
+use rand::prelude::*;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct NodePool {
+    prepared_metadata: HashMap<Vec<u8>, PreparedMetadata>,
+    token_map: TokenMap,
+    pub nodes: Vec<CassandraNode>,
+}
+
+impl NodePool {
+    pub fn new(nodes: Vec<CassandraNode>) -> Self {
+        Self {
+            token_map: TokenMap::new(nodes.as_slice()),
+            nodes,
+            prepared_metadata: HashMap::new(),
+        }
+    }
+
+    pub fn set_nodes(&mut self, nodes: Vec<CassandraNode>) {
+        self.nodes = nodes;
+        self.token_map = TokenMap::new(self.nodes.as_slice());
+    }
+
+    pub fn add_prepared_result(&mut self, id: Vec<u8>, metadata: PreparedMetadata) {
+        self.prepared_metadata.insert(id, metadata);
+    }
+
+    pub fn random_node(&mut self, rng: &mut SmallRng) -> &mut CassandraNode {
+        self.nodes
+            .iter_mut()
+            .filter(|x| x.is_up)
+            .choose(rng)
+            .unwrap()
+    }
+
+    pub fn get_random_node_in_dc_rack(
+        &mut self,
+        rack: &String,
+        rng: &mut SmallRng,
+    ) -> &CassandraNode {
+        self.nodes
+            .iter()
+            .filter(|x| x.rack == *rack && x.is_up)
+            .choose(rng)
+            .unwrap()
+    }
+
+    /// Get a token routed replica node for the supplied execute message (if exists)
+    pub fn replica_node(
+        &mut self,
+        execute: &BodyReqExecuteOwned,
+        version: &Version,
+    ) -> Option<&mut CassandraNode> {
+        let metadata = self
+            .prepared_metadata
+            .get(&execute.id.clone().into_bytes().unwrap())
```

Comment:
We can avoid the clone here by just directly storing the CBytesShort.
![image](https://user-images.githubusercontent.com/5120858/192920707-1bf7b2be-bb37-4db3-834f-1058142565b7.png)